DM-11790 Move jointcal document out of jointcal and into technote

parejkoj opened the pull request.

Commit:
Move jointcal document out of jointcal and into technote

Update technote files to match new TeX usage (instead of RST).

Add references as bibtex

jonathansick committed:
Add credentials for Lander deployment

timj commented:
Your commit author emails are messed up so that GitHub doesn't know who you are.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+stack.}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% User specified LaTeX commands.
+\newcommand{\ClName}[1]{{\bf #1}}
```
Please live in the 1990s in a world where latex2e exists.

parejkoj replied:
I have no idea what you mean by that. The `newcommand` blocks here were in the original document from Pierre. What should I use instead?

timj replied:
`\textbf` not `{\bf`

parejkoj replied:
I've removed both of those commands and just used the `\code{}` and `\method{}` commands from lsst-texmf.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+\author{John Parejko (University of Washington), Pierre Astier (LPNHE/IN2P3/CNRS
+Paris)}
+\setDocRef{DMTN-036}
+\date{\today}
```
When you release and tag this, fix the date.

parejkoj replied:
What does "release" count as? I'll merge to master, but this is by no means "complete".

timj replied:
Release probably means "when the JIRA ticket is done". You don't want a tweak to the build system next year to change the document date.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+\usepackage{xunicode}
+\begin{document}
+\maketitle
+\tableofcontents{}
```
Remove this line. You have a second empty table of contents in the PDF.

parejkoj replied:
Didn't know that ToC was provided by lsst-texmf.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+
+\makeatother
+
+\usepackage{xunicode}
```
This package is already loaded by the class.

parejkoj replied:
That's an auto-insert by LyX. I'll ask them about adding it to the list of things that can be defined as class-provided in the layout.

parejkoj replied:
Upstream request sent. I'll leave this as is for now (it doesn't break the build).

parejkoj replied:
Upstream response is that this will be a configurable in the layout in LyX 2.3.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+as our reference catalog, and may supplement it with deeper data (e.g.
+PS1) where available.
+
+SCAMP \footnote{See \href{http://www.astromatic.net/software/scamp}{http://www.astromatic.net/software/scamp}}is
```
Cite SCAMP (I really don't like footnote URL citations).

parejkoj replied:
What's the preferred reference for SCAMP?

timj replied:
https://ui.adsabs.harvard.edu/#abs/2006ASPC..351..112B/abstract

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+clear that one cannot fit simultaneously the position on the sky and
+the mappings from CCD coordinates to the sky, without extra constraints:
+the ``sky'' coordinate system is then undefined, and one needs reference
+positions in order to fully define this frame. We use the GAIA catalog
```
I believe you are using the Gaia catalog not the GAIA catalog.

parejkoj replied:
What is the best reference for Gaia DR1?

timj replied:
No idea. https://ui.adsabs.harvard.edu/#abs/2016A%26A...595A...2G/abstract ?

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+absolute photometry.
+
+The plan of this note is as follows: we first sketch the algorithm
+(\S\ref{sec:algo}). We provide our least-squares formulation in
```
Consider `\secref`.

parejkoj replied:
I've switched to using formatted references with prettyref, which handles it automatically.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+\S\ref{sec:mathematical-formalism}, and describe the how we evaluate
+the derivatives with respect to the parameters. We then describe how
+we associate the measurements of each object in different exposures
+in \S \ref{sec:catalog-association}. 
```
Should not be extra space here.

parejkoj commented:
Deleting my previous comment: somehow my laptop .gitconfig lost its email address. I'll rebase again and push.

parejkoj commented:
Although github still doesn't realize that my laptop and desktop are both me (they both have the same user email configured). Any ideas there are welcome.

timj commented on DMTN-036.tex:
```diff
@@ -0,0 +1,808 @@
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% User specified LaTeX commands.
+\newcommand{\ClName}[1]{{\bf #1}}
+\newcommand{\RoutineName}[1]{\texttt{#1}}
```
There is `\code{}`

parejkoj replied:
Ah, the one defined in lsst-texmf? It's not described here that I can see: http://lsst-texmf.lsst.io/

There's a whole bunch of macros defined in lsst-texmf, but I don't know what they're each for, though I think I can guess (e.g., `\code{}`, `\class{}` (which are defined in different sections, `\method{}`).

timj replied:
Yes. I haven't worked out how to document them all without duplicating everything.

parejkoj replied:
I'm using `code` and `method` now.

timj commented on DMTN-036.tex:
Use the `{lstlisting}` environment here.

parejkoj replied:
I'm removing that whole section, since there's a `jointcalTask` now.

timj commented on DMTN-036.tex:
AST also uses radians.

parejkoj replied:
I'm not sure about the utility of this whole section, so I'm going to leave it for now.

timj commented on DMTN-036.tex:
"Fortran"

timj commented on DMTN-036.tex:
Should "jointcal" be `\textsc{jointcal}` or something?

parejkoj replied:
I don't know. Is there a `\package{}` or equivalent defined in lsst-texmf?

timj replied:
I don't think there is. We could add one. This is my MNRAS style guide persona kicking in.

parejkoj replied:
There is `\pack{}`, which I assume means `\package{}` (and which I'd suggest we rename to that...)

parejkoj replied:
I've used `\code` for all of these, which does the job.

timj commented on DMTN-036.tex:
That empty `\emph` seems wrong.

timj reviewed (approved):
Looks like a good start. Haven't read through in detail but I have a few fixes that are needed.

timj commented:
```
commit b30332deaf0335b7c1616dee5a8b22670676e77c
Author: John K. Parejko <parejkoj@fiddlercrab.local>
Date:   Tue Sep 5 11:51:52 2017 -0700
```
Note `.local`.

parejkoj committed:
Post-review cleanups

Remove ref that was added to lsst-texmf/refs_ads
added GAIA and SCAMP refs

Cleanup references.
